internal/olm/operator/catalog_source_test.go

```diff
@@ -0,0 +1,130 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package olm
+
+import (
+	"context"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+
+	olmclient "github.com/operator-framework/operator-sdk/internal/olm/client"
+	internalregistry "github.com/operator-framework/operator-sdk/internal/olm/operator/internal"
+	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+// newFakeClientSet() returns a fake kubernetes clientset
+func newFakeClientSet() kubernetes.Interface {
+	return fake.NewSimpleClientset()
+}
+
+func TestCatalogSource(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Catalog Source Suite")
+}
+
+var _ = Describe("CatalogSource", func() {
+
+	Describe("creating catalog source", func() {
+
+		Context("when using valid registry pod values and a mock client", func() {
+			expectedCatSrcName := "quay-io-mock-example-operator-bundle-0-1-0-ocs"
+			expectedNamespace := "olm"
+			expectedCatSrcKind := "CatalogSource"
+			expectedDisplayName := "quay-io-mock-example-operator-bundle-0-1-0"
+			expectedPublisher := "operator-sdk"
+
+			var rp *internalregistry.RegistryPod
+			var err error
+
+			BeforeEach(func() {
+				rp, err = internalregistry.NewRegistryPod(newFakeClientSet(),
+					"/database/index.db", "quay.io/mock/example-operator-bundle:0.1.0", "olm")
+				Expect(err).To(BeNil())
+				Expect(rp).NotTo(BeNil())
+			})
+
+			It("should initialize the CatalogResources struct successfully", func() {
+				// Call initialize using a registry pod
+				mockCatResources, err := initialize(rp)
+				Expect(err).To(BeNil())
+				Expect(mockCatResources).NotTo(BeNil())
+			})
+
+			It("should create a catalog source successfully", func() {
+				mockCatResources, err := initialize(rp)
+				Expect(err).To(BeNil())
+
+				// Call CreateCatalogSource using a fake controller runtime client
+				cs, err := mockCatResources.CreateCatalogSource(context.Background(),
+					&olmclient.Client{
+						KubeClient: fakeclient.NewFakeClient(),
+					})
+
+				Expect(err).To(BeNil())
+				Expect(cs).NotTo(BeNil())
+				Expect(cs.Name).To(Equal(expectedCatSrcName))
+				Expect(cs.ObjectMeta.Namespace).To(Equal(expectedNamespace))
+				Expect(cs.Kind).To(Equal(expectedCatSrcKind))
+				Expect(cs.Spec.DisplayName).To(Equal(expectedDisplayName))
+				Expect(cs.Spec.Publisher).To(Equal(expectedPublisher))
+
```

Comment:
IMO the initialize do not fit well and we just need 1 method to create a new struct. 

```suggestion
			It("should create a catalog source successfully", func() {
				// Call CreateCatalogSource using a fake controller runtime client
				cs, err := mockCatResources.CreateCatalogSource(rp, context.Background(),
					&olmclient.Client{
						KubeClient: fakeclient.NewFakeClient(),
					})

				Expect(err).To(BeNil())
				Expect(cs).NotTo(BeNil())
				Expect(cs.Name).To(Equal(expectedCatSrcName))
				Expect(cs.ObjectMeta.Namespace).To(Equal(expectedNamespace))
				Expect(cs.Kind).To(Equal(expectedCatSrcKind))
				Expect(cs.Spec.DisplayName).To(Equal(expectedDisplayName))
				Expect(cs.Spec.Publisher).To(Equal(expectedPublisher))

```